### PR TITLE
Makefile.calicoctl: disable "-i" for go build for non-linux

### PR DESCRIPTION
**Pull request by tomdee**

### CODE REVIEW

File: Makefile.calicoctl [hunk 1]
```diff
@@ -67,7 +67,12 @@ $(CTL_CONTAINER_CREATED): calicoctl/Dockerfile.calicoctl dist/calicoctl
 
 ## Build calicoctl
 binary: $(CALICOCTL_FILES) vendor
-	GOOS=$(OS) GOARCH=$(ARCH) CGO_ENABLED=0 go build -v -i -o dist/calicoctl-$(OS)-$(ARCH) $(LDFLAGS) "./calicoctl/calicoctl.go"
+  # Don't try to "install" the intermediate build files (.a .o) when not on linux
+	# since there are no write permissions for them in our linux build container.
+  if [ "$(GOOS)" == "linux" ]; then \
```
Comment (caseydavenport) [inline Makefile.calicoctl]: Where is GOOS set? Do you mea the "OS" variable instead?

File: Makefile.calicoctl [hunk 2]
```diff
@@ -67,7 +67,12 @@ $(CTL_CONTAINER_CREATED): calicoctl/Dockerfile.calicoctl dist/calicoctl
 
 ## Build calicoctl
 binary: $(CALICOCTL_FILES) vendor
-	GOOS=$(OS) GOARCH=$(ARCH) CGO_ENABLED=0 go build -v -i -o dist/calicoctl-$(OS)-$(ARCH) $(LDFLAGS) "./calicoctl/calicoctl.go"
+  # Don't try to "install" the intermediate build files (.a .o) when not on linux
```
Comment (robbrockbank) [inline Makefile.calicoctl]: Mix of spaces and tabs here.

**Comment by robbrockbank** [comment]
Probably me not understanding the -i flag correctly ... but given the comment, should the -i flag be omitted for all builds?

**Comment by robbrockbank** [comment]
LGTM

**Pull request merged**